src/webots/gui/WbMainWindow.cpp

```diff
@@ -1615,6 +1629,138 @@ void WbMainWindow::exportHtml() {
 
   WbSimulationState::instance()->setMode(currentMode);
 }
+void WbMainWindow::ShareMenu() {
+  WbSimulationState::Mode currentMode = WbSimulationState::instance()->mode();
+  WbShareWindow *shareWindowUi = new WbShareWindow(this);
+  shareWindowUi->exec();
+  WbSimulationState::instance()->setMode(currentMode);
+}
+
+void WbMainWindow::uploadScene() {
+  WbWorld *world = WbWorld::instance();
+  world->exportAsHtml(WbStandardPaths::webotsTmpPath() + "export_cloud.html", false);
+  sendCloudRequest();
+}
+
+void WbMainWindow::sendCloudRequest() {
+  QNetworkRequest request(QUrl("https://beta.webots.cloud/ajax/animation/create.php"));
+  QHttpMultiPart *multiPart = new QHttpMultiPart(QHttpMultiPart::FormDataType);
+  bool animation = false;
+
+  QStringList filenames = QDir(WbStandardPaths::webotsTmpPath() + "textures/")
+                            .entryList(QStringList() << "*.jpg"
+                                                     << "*.JPG"
+                                                     << "*.jpeg"
+                                                     << "*.png"
+                                                     << "*.hdr",
+                                       QDir::Files);
+  if (filenames.isEmpty())  // add empty texture
+    filenames.append("");
+
+  if (QFileInfo(WbStandardPaths::webotsTmpPath() + "export_cloud.json").exists()) {
+    filenames << "export_cloud.json";
+    animation = true;
+  }
+  filenames << "export_cloud.x3d";
+
+  // add files content
+  QMap<QString, QString> map;
+  foreach (QString filename, filenames) {
+    QHttpPart mainPart;
+    if (filename.contains("x3d")) {
+      map["foldername"] = WbStandardPaths::webotsTmpPath();
+      map["name"] = "scene-file";
+    } else if (filename.contains("json")) {
+      map["foldername"] = WbStandardPaths::webotsTmpPath();
+      map["name"] = "animation-file";
+    } else {
+      map["foldername"] = WbStandardPaths::webotsTmpPath() + "textures/";
+      map["name"] = "textures[]";
+    }
+
+    mainPart.setHeader(QNetworkRequest::ContentDispositionHeader,
+                       QVariant("form-data; name=" + map["name"] + "; filename=" + filename));
+
+    // read file content
+    if (filename != "") {
+      QFile *file = new QFile(map["foldername"] + filename);
+      file->open(QIODevice::ReadOnly);
+
+      mainPart.setBodyDevice(file);
+      file->setParent(multiPart);
+    }
+    multiPart->append(mainPart);
+  }
+  // add other information
+  QMap<QString, QString> infos;
+  infos["type"] = "S";
+  if (animation) {
+    infos["type"] = "A";
+  }
+  infos["user"] = "null";
+  infos["password"] = "null";
+
+  QMapIterator<QString, QString> iteInfos(infos);
```

Comment:
What does "ite" stand for? Can we choose a better name?
```suggestion
  QMapIterator<const QString, const QString> iteInfos(infos);
```